client/index.js

```diff
@@ -206,10 +275,174 @@ window.onload = () => {
+  'Quarry': quarryValid,
+  'Bank': bankValid,
+  'Embassy': embassyValid,
+  'Guard Tower': guardTowerValid,
+  'Customs Office': customsOfficeValid,
+  'Casino': casinoValid,
+  'Lighthouse': lightHouseValid,
+  'Graveyard': graveyardValid,
+  'Jail': jailValid,
+  'Workshop': workShopValid,
+  'Mill': millValid,
+  'Wall': wallValid,
+  'Church': churchValid,
+  'Boulevard': boulevardValid,
+  'Aqueduct': aqueductValid,
+  'Harbor': harborValid,
+  'Shipyard': shipyardValid,
+  'Trolley': trolleyValid,
+  'City Hall': cityHallValid,
+  'Bridge': bridgeValid,
+  'Tunnel': tunnelValid,
+  'Sewers': sewersValid,
+  'Depot': depotValid,
+  'Plaza': plazaValid,
+}
+
+function harborValid(coords) {
+  if (coords.length != 1) {
+    return false
+  } else { 
+    row = coords[0][0]
+    col = coords[0][1]
+    if (tileAdjacentToFriendlyOrPlacement(row, col) && BOARD[row][col].type === 'w') {
+      return true
+    } else {
+      return false
+    }
+  }
+}
+
+function harborValid() {return false}
+function taxHouseValid() {return false}
+function docksValid() {return false}
+function settlementValid() {return false}
+function bazaarValid() {return false}
+function quarryValid() {return false}
+function bankValid() {return false}
+function embassyValid() {return false}
+function guardTowerValid() {return false}
+function customsOfficeValid() {return false}
+function casinoValid() {return false}
+function lightHouseValid() {return false}
+function graveyardValid() {return false}
+function jailValid() {return false}
+function workShopValid() {return false}
+function millValid() {return false}
+function wallValid() {return false}
+function churchValid() {return false}
+function boulevardValid() {return false}
+function aqueductValid() {return false}
+function harborValid() {return false}
+function shipyardValid() {return false}
+function trolleyValid() {return false}
+function cityHallValid() {return false}
+function bridgeValid() {return false}
+function tunnelValid() {return false}
+function sewersValid() {return false}
+function depotValid() {return false}
+function plazaValid() {return false}
+
+// get all adjacent tile coordinates
+function getAdjacentCoordinates(row, col) {
+  adjacents = [[row, col-1], [row, col+1]]
+	if (row%2 == 0) {
+    adjacents.push([row-1, col-1])
+    adjacents.push([row+1, col-1])
+    adjacents.push([row-1, col])
+    adjacents.push([row+1, col])
+	} else {
+    adjacents.push([row-1, col])
+    adjacents.push([row+1, col])
+    adjacents.push([row-1, col+1])
+    adjacents.push([row+1, col+1])
+  }
+  return adjacents
+}
+
+// check if tile at coordinates is friendly
+function friendly(row, col) {
+	try {
+		if (STARTING_PLAYER) {
+  		return BOARD[row][col].marker == 'player_one'
+  	} else {
+  		return BOARD[row][col].marker == 'player_two'
+  	}
+	} catch(e) {
+		return false
+	}
+}
+
+// check if tile at coordinates is adjacent to friendly marker
+function tileAdjacentToFriendly(row, col) {
```

Comment:
You can combine tileAdjacentToFriendly() and tileAdjacentToFriendlyOrPlacement() by keeping track of 2 booleans for the two checks, and returning both. The callers can pick which they want. For clarity, return those as a little object of the two instead of an array.